src/services/dmr/DmrService.ts

```diff
@@ -179,17 +179,45 @@ export class DmrService {
     let msg: string = "An unspecified error has been received from the server";
     if (response.status !== 401) {
       let result: any = response.data;
-      if (result && result["failure-description"] != null) {
-        if (isNotNullOrUndefined(result["failure-description"]["domain-failure-description"])) {
-          msg = result["failure-description"]["domain-failure-description"];
-        } else {
-          msg = result["failure-description"];
+      if (isNotNullOrUndefined(result)) {
+        if (isNotNullOrUndefined(result["failure-description"])) {
+          if (isNotNullOrUndefined(result["failure-description"]["domain-failure-description"])) {
+            msg = result["failure-description"]["domain-failure-description"];
+          } else {
+            msg = result["failure-description"];
+          }
+        }
+        // see if we can find the cause of this error
+        let cause: string = this.findErrorCause(result);
+        if (isNotNullOrUndefined(cause)) {
+          msg = cause;
```

Comment:
This always overrides `msg` if it has previously been set. Why not look for the cause first. If not null, great you have your msg, otherwise check the failure description.

Reply:
+1 ok, makes sense